File: ios/framework-objc/liferay-mobile-widgets/common/LiferayContext.m
//
//  LiferayContext.m
//  liferay-mobile-portlets-sample
//
//  Created by jmWork on 09/06/14.
//  Copyright (c) 2014 Liferay. All rights reserved.
//

#import "LiferayContext.h"
#import "LoginStorage.h"
#import "LRSession.h"

@implementation LiferayContext


+ (instancetype)sharedInstance {
	static LiferayContext *_instance;
    static dispatch_once_t onceToken;

    dispatch_once(&onceToken, ^{
        _instance = [[self alloc] init];
    });

    return _instance;
}


- (id)init {
	if (self = [super init]) {
		self.currentSession = [LoginStorage loadStoredSession];

		NSString *propertiesPath = [[NSBundle mainBundle] pathForResource:@"liferay-context" ofType:@"plist"];
		if (propertiesPath) {
			NSDictionary *properties = [[NSDictionary alloc] initWithContentsOfFile:propertiesPath];
			self.server = properties[@"server"];
			self.companyId = [properties[@"companyId"] longLongValue];
			self.groupId = [properties[@"groupId"] longLongValue];
		}
	}

	return self;
}

- (NSString*) currentLocale {
	NSString *locale = [[NSLocale autoupdatingCurrentLocale] localeIdentifier];
	NSRange startRange = [locale rangeOfString:@"_"];
	return [locale stringByReplacingCharactersInRange:NSMakeRange(0, startRange.length+1) withString:[[NSLocale preferredLanguages] objectAtIndex:0]];
}

@end

File: ios/framework/liferay-ios-widgets/Base/LiferayContext.swift
//
//  LiferayContext.swift
//  liferay-swift-widgets
//
//  Created by jmWork on 01/07/14.
//  Copyright (c) 2014 Liferay. All rights reserved.
//

import Foundation

struct LiferayContext {

	var server:String = "http://localhost:8080"
	var companyId:Int = 10154
	var groupId:Int = 10181

	var currentSession:LRSession?

	static var instance = LiferayContext()

	init() {
		let propertiesPath = NSBundle.mainBundle().pathForResource("liferay-context", ofType:"plist")
		if propertiesPath {
			loadContextFile(propertiesPath)
		}
		else {
			println("WARNING: liferay-context.plist file is not found. Falling back to template liferay-context-sample.list")

			if let templatePath = NSBundle.mainBundle().pathForResource("liferay-context-sample", ofType:"plist") {
				loadContextFile(propertiesPath)
			}
			else {
				println("ERROR: liferay-context-sample.plist file is not found")
			}
		}
	}

	mutating func loadContextFile(propertiesPath:String) {
		let properties = NSDictionary(contentsOfFile: propertiesPath)
		server = properties["server"] as String;
		companyId = properties["companyId"] as Int
		groupId = properties["groupId"] as Int
	}

	mutating func createSession(username:String, password:String) -> LRSession {
		self.currentSession = LRSession(server, username:username, password:password)
		return self.currentSession!
	}

	mutating func clearSession() {
		self.currentSession = nil
	}

}
